tests/ballerina-compiler-api-test/src/test/resources/test-src/symbols/resource_method_symbols.bal

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+client class BookService {
+    resource function get .() returns Book[] {
+        return [
+            {id: 1, title: "AtoZ", author: "Jeo"},
+            {id: 2, title: "BtoY", author: "Frank"}
+        ];
+    }
+
+    resource function get authors/[string... names]() returns string[] {
+        return names;
+    }
+
+    resource function post authors(string author) returns string {
+        return author;
+    }
+
+    resource function get registered/[float|decimal isbn]/[int... ex]() returns Book? {
```

Comment:
Will this address if we have a path parameter union type with pre-defined(typeReference)?
ex: 
``` ballerina
public type Isbn float|decimal;
resource function get registered/[Isbn isbn]() returns Book? {}
```

Reply:
Yes its covered